packages/metadata-common/src/MetadataWidget.tsx

```diff
@@ -152,13 +179,111 @@ export class MetadataDisplay<
     );
   }
 
+  filteredMetadata = (searchValue: string, filterTags: string[]): void => {
+    // filter with search
+    let filteredMetadata = this.props.metadata.filter(
+      (metadata: IMetadata, index: number, array: IMetadata[]): boolean => {
+        return (
+          metadata.name.toLowerCase().includes(searchValue.toLowerCase()) ||
+          metadata.display_name
+            .toLowerCase()
+            .includes(searchValue.toLowerCase())
+        );
+      }
+    );
+
+    // filter with tags
+    if (filterTags.length !== 0) {
+      filteredMetadata = filteredMetadata.filter(metadata => {
+        return filterTags.some(tag => {
+          if (metadata.metadata.tags) {
+            return metadata.metadata.tags.includes(tag);
+          }
+          return false;
+        });
+      });
+    }
+
+    this.setState({
+      metadata: filteredMetadata,
+      searchValue: searchValue,
+      filterTags: filterTags
+    });
+  };
+
+  getActiveTags(): string[] {
+    const tags: string[] = [];
+    for (const metadata of this.props.metadata) {
+      if (metadata.metadata.tags) {
+        for (const tag of metadata.metadata.tags) {
+          if (!tags.includes(tag)) {
+            tags.push(tag);
+          }
+        }
+      }
+    }
+    return tags;
+  }
+
+  static matchesSearch(
+    searchValue: string,
+    filterTags: Set<string>,
+    metadata: IMetadata
+  ): boolean {
+    // True if search string is in name, display_name, or language of snippet
+    // or if the search string is empty
+    const matchesSearch =
+      metadata.name.toLowerCase().includes(searchValue) ||
+      metadata.display_name.toLowerCase().includes(searchValue);
+    // True if there are no tags selected or if there are tags that match
+    // tags of metadata
+    const matchesTags =
+      filterTags.size === 0 ||
+      (metadata.metadata.tags &&
+        metadata.metadata.tags.some((tag: string) => filterTags.has(tag)));
+    return matchesSearch && matchesTags;
+  }
+
+  static getDerivedStateFromProps(
```

Comment:
this can't be `static` if we want to override the `matchesSearch` call found in this function in child classes